src/algorithms/local_search/local_search.cpp

```diff
@@ -758,6 +771,16 @@ void LocalSearch<Route,
             continue;
           }
 
+          const auto& s_v = _input.vehicles[_sol[s_t.first].vehicle_rank];
+          const auto& t_v = _input.vehicles[_sol[s_t.second].vehicle_rank];
+
+          if (s_rank - t_rank + _sol[s_t.second].size() >
+                s_v.max_number_of_tasks or
+              t_rank - s_rank + _sol[s_t.first].size() >
+                t_v.max_number_of_tasks) {
+            continue;
```

Comment:
Not a good idea on second though because it would overflow if `max_number_of_tasks` is `std::numeric_limits<size_t>::max()`. Maybe putting braces around `(_sol[s_t.second].size() - t_rank)`, guaranteed to be non-negative?